pkg/protocol/stan/write_message.go

```diff
@@ -0,0 +1,45 @@
+package stan
+
+import (
+	"context"
+	"github.com/cloudevents/sdk-go/pkg/binding"
+	"github.com/cloudevents/sdk-go/pkg/binding/format"
+	"io"
+)
+
+// Fill the provided writer with the bindings.Message m.
+// Using context you can tweak the encoding processing (more details on binding.Write documentation).
+func WriteMsg(ctx context.Context, m binding.Message, writer io.ReaderFrom, transformers binding.TransformerFactories) error {
+	structuredWriter := &stanMessageWriter{writer}
+
+	_, err := binding.Write(
+		ctx,
+		m,
+		structuredWriter,
+		nil,
+		transformers,
+	)
+	return err
+}
+
+type stanMessageWriter struct {
+	io.ReaderFrom
+}
+
+func (w *stanMessageWriter) SetStructuredEvent(ctx context.Context, f format.Format, event io.Reader) error {
+	if _, err := w.ReadFrom(event); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (w *stanMessageWriter) Start(ctx context.Context) error {
```

Comment:
You don't need this to implement `StructuredWriter`

Reply:
I copied this from nats, should I update both protocols?

Reply:
yes thank you! `Start` and `End` belongs only to `BinaryWriter`